src/app/components/fruit-table/fruit-table.component.ts

```diff
@@ -1,19 +1,109 @@
-import { Component, OnInit } from '@angular/core';
-import {FruitTableViewModel} from './fruit-table-view-model';
+import { Component, Inject, OnInit, ViewEncapsulation } from "@angular/core";
+import { FruitTableViewModel } from "./fruit-table-view-model";
+import {
+  MatDialog,
+  MatDialogRef,
+  MAT_DIALOG_DATA,
+} from "@angular/material/dialog";
+import { Fruit } from "src/app/models/fruit";
 
 @Component({
-  selector: 'app-fruit-table',
-  templateUrl: './fruit-table.component.html',
-  styleUrls: ['./fruit-table.component.scss'],
-  providers: [FruitTableViewModel]
+  selector: "app-fruit-table",
+  templateUrl: "./fruit-table.component.html",
+  styleUrls: ["./fruit-table.component.scss"],
+  providers: [FruitTableViewModel],
 })
 export class FruitTableComponent implements OnInit {
-  columnsToDisplay = ['id', 'name', 'genus', 'calories'];
+  columnsToDisplay = [
+    "id",
+    "name",
+    "genus",
+    "calories",
+    "carbohydrates",
+    "sugar",
+  ];
 
-  constructor(public viewModel: FruitTableViewModel) {
+  options = [
+    { value: "nameAsc", viewValue: "Name Ascending" },
+    { value: "nameDes", viewValue: "Name Descending" },
+    { value: "carbsAsc", viewValue: "Carbohydrates Ascending" },
+    { value: "carbsDes", viewValue: "Carbohydrates Descending" },
+  ];
 
+  fruit: Fruit;
+
+  constructor(
+    public viewModel: FruitTableViewModel,
+    public dialog: MatDialog
+  ) {}
+
+  ngOnInit(): void {}
+
+  applyFilter(event: string) {
+    let searchTerm = event.toLowerCase();
+    this.viewModel.filteredData$.next(
+      this.viewModel.fruitData$.value.filter(
+        (fruit) =>
+          fruit.genus.toLowerCase().includes(searchTerm) ||
+          fruit.name.toLowerCase().includes(searchTerm) ||
+          fruit.family.toLowerCase().includes(searchTerm) ||
+          fruit.order.toLowerCase().includes(searchTerm)
+      )
+    );
   }
+  onSelectEvent(event: string) {
```

Comment:
I think a switch statement might be a little better choice here

Reply:
This was done to the challenge specs but often we sort on column click, can you change the sorting to work when the title is clicked?  Will need some icon to indicate direction and also to change the cursor when hovering over the sort button.